tests/integration/header/header_test.go

```diff
@@ -42,6 +43,28 @@ func (dispatcher) ExecuteTemplate(rw http.ResponseWriter, t safehttp.Template, d
 	return nil
 }
 
+type responseWriter struct {
+	header http.Header
+	writer io.Writer
+	status int
+}
+
+func newResponseWriter(w io.Writer) *responseWriter {
+	return &responseWriter{header: http.Header{}, writer: w, status: http.StatusOK}
+}
+
+func (r responseWriter) Header() http.Header {
```

Comment:
Keep the receiver type (pointer or not) the same across all methods.

Reply:
Fixed. But why do you want this? Having just a normal receiver sometimes signals that the function is read-only.

Reply:
Well, not exactly.

Value receivers are used for passing objects *as values* (copies). Having a value receiver doesn't prevent you from altering objects that `responseWriter` references (e.g. slices, maps, pointers). So it's at best a "shallow" read-only.

Reply:
Right!